### ISSUE TITLE
Teleinfo 1 Total accepts 0 values (noise in measurement), which performs a reset of the counter, ending up with unreliable (GIANT!) consumption 

### ISSUE THREAD

**Issue by edelans** [Aohzan/ecodevices#13]
Hello again, 

I'm fighting with the energy dashboard configuration. 

My energy usage looks like this in the dashbaord : 

![CleanShot 2021-09-09 at 13 15 15](https://user-images.githubusercontent.com/2805640/132676209-fbc8aa7a-2586-42ac-96d2-540a42d0ce0d.png)

The bars are way too high ! Actually, 12,448 kWh you see in the chart was the reading of my "compteur linky" at that time. 

This is because, when you have a closer look at the teleinfo 1 total entity, you see there is a 0kWh sent at some point : 

![CleanShot 2021-09-09 at 13 17 02](https://user-images.githubusercontent.com/2805640/132676486-112dd153-6f83-41f6-b4ba-89f770b26dfd.png)

with the `total_increasing` `state_class`, what happens is that the energy dashboard thinks I have consummed the 12MWh at that moment, making all other bars of the chart ~0 : 

![CleanShot 2021-09-09 at 13 20 12](https://user-images.githubusercontent.com/2805640/132676911-d512817c-fd50-4ad8-8625-1db60a95777a.png)

I guess the integration should filter out `0` values (or even filter out decreasing values ?) as [suggested in the doc](https://developers.home-assistant.io/blog/2021/08/16/state_class_total/) : 

![CleanShot 2021-09-09 at 13 18 30](https://user-images.githubusercontent.com/2805640/132676624-8f70fcfd-cb7b-44cc-893b-f5610c267c50.png)

![CleanShot 2021-09-09 at 13 22 13](https://user-images.githubusercontent.com/2805640/132677191-4c19e518-1968-4f8b-ae50-8bf0a8999d18.png)

I'd be happy to help, I know Python,  but I've never dev'd any HA plugin. I guess the filtering should be done around those lines https://github.com/Aohzan/ecodevices/blob/3fc98c1308f73836c7e40bd21d23a0c5d8376b0a/custom_components/ecodevices/sensor.py#L202 ?

**Comment by edelans** [comment]
I think I found a workaround by defining a new sensor in my `configuration.yaml` file. It filters out decreasing values.

```
  - sensor:
      name: Elec Consumption Wh increasing only
      state: >
        {% if states('sensor.teleinfo_1_total') > states('sensor.elec_consumption_wh_increasing_only') %}
          {{ states('sensor.teleinfo_1_total') | float(states('sensor.elec_consumption_wh_increasing_only' )) }}
        {% else %} 
          {{ states('sensor.elec_consumption_wh_increasing_only') }}
        {% endif %}
      icon: mdi:flash
      unit_of_measurement: "Wh"
      state_class: "total_increasing"
      device_class: "energy"
```

**Comment by Aohzan** [comment]
Thanks for the report, I will try to add the anomaly detection in the integration

**Comment by Aohzan** [comment]
I couldn't check if the new value is greater than previous, so I only excluded the value if it's 0
You can test the release here : https://github.com/Aohzan/ecodevices/tree/2.5.0b

**Comment by edelans** [comment]
Hello @Aohzan ! Super cool you managed to fix the 2 issues that fast !

What do you recommend to review this ? (I use HACS to import you plugin, and I configured it with the GUI). Should I just temporarily replace the hacs version of the add-on with https://github.com/Aohzan/ecodevices/tree/2.5.0b ? I don't want to mess around with my prod environment 😅

reminds me of that tweet : https://twitter.com/stahnma/status/634849376343429120?lang=fr 

![CleanShot 2021-09-16 at 23 11 47](https://user-images.githubusercontent.com/2805640/133686225-afcb8745-c906-438f-93e9-69dd9fcc1e5a.png)



Unfortunately, I have only one HA instance !

**Comment by Aohzan** [comment]
You can test on your main HA, no issue on my side

**Comment by Aohzan** [comment]
fix in last 3.0.0 version
